File: backend_java/src/main/java/br/com/fiap/chronosdtn/model/Link.java
package br.com.fiap.chronosdtn.model;

// Por que: Importa as anotações do Jakarta Persistence para mapear chaves e relacionamentos de rede.
import jakarta.persistence.*;
// Por que: Importa Lombok para automatizar escrita de getters, setters e construtores de classe.
import lombok.*;
// Por que: Permite carregar carimbos de data/hora de precisão espacial (microsegundos) nas janelas de rádio.
import java.time.LocalDateTime;

// Por que: Declara que esta classe mapeia uma tabela do banco de dados relacional.
@Entity
// Por que: Associa esta classe à tabela física de enlaces espaciais.
@Table(name = "TB_CHRONOS_LINK")
// Por que: Lombok - Métodos getter gerados automaticamente.
@Getter
// Por que: Lombok - Métodos setter gerados automaticamente.
@Setter
// Por que: Lombok - Construtor padrão com todos os campos de link.
@AllArgsConstructor
// Por que: Lombok - Construtor vazio para o Hibernate.
@NoArgsConstructor
public class Link {

    // Por que: Define a chave primária de enlace de rede orbital.
    @Id
    @Column(name = "ID_LINK")
    private Long id;

    // Por que: Relacionamento de muitos-para-um ligando ao nó espacial de origem do sinal.
    @ManyToOne(fetch = FetchType.EAGER)
    // Por que: Mapeia a coluna da chave estrangeira referenciando o ID do nó transmissor.
    @JoinColumn(name = "ID_NODE_SOURCE", nullable = false)
    private Node sourceNode;

    // Por que: Relacionamento de muitos-para-um ligando ao nó receptor do sinal.
    @ManyToOne(fetch = FetchType.EAGER)
    // Por que: Mapeia a chave estrangeira referenciando o ID do nó destinatário.
    @JoinColumn(name = "ID_NODE_DEST", nullable = false)
    private Node destNode;

    // Por que: Início da disponibilidade da janela orbital de comunicação rádio.
    @Column(name = "DT_START", nullable = false)
    private LocalDateTime startTime;

    // Por que: Fim da disponibilidade da janela orbital de comunicação rádio.
    @Column(name = "DT_END", nullable = false)
    private LocalDateTime endTime;

    // Por que: Banda útil do enlace de rádio em Kilobits por segundo.
    @Column(name = "VL_BANDWIDTH_KBPS", nullable = false)
    private Double bandwidthKbps;

    // Por que: Latência de atraso do sinal baseada na velocidade da luz (segundos).
    @Column(name = "VL_LATENCY_SECONDS", nullable = false)
    private Double latencySeconds;

    // Por que: Estado atual de rádio da janela (ex: UP, DOWN, SCHEDULED).
    @Column(name = "ST_LINK", nullable = false, length = 20)
    private String status;
}

File: backend_net/ChronosDtnNetApi/Models/Link.cs
using System;

namespace ChronosDtnNetApi.Models;

// Por que: Define a janela de contato físico (link de rádio) entre duas estações no espaço.
public class Link
{
    // Por que: Chave primária do registro do enlace de comunicação.
    public long Id { get; set; }

    // Por que: Referência estrangeira do nó de origem da transmissão.
    public long SourceNodeId { get; set; }

    // Por que: Objeto do nó emissor (carregamento por navegação ORM).
    public Node? SourceNode { get; set; }

    // Por que: Referência estrangeira do nó destinatário da transmissão.
    public long DestNodeId { get; set; }

    // Por que: Objeto do nó receptor (carregamento por navegação ORM).
    public Node? DestNode { get; set; }

    // Por que: Instante de início em que a janela orbital permite contato direto de rádio.
    public DateTime StartTime { get; set; }

    // Por que: Instante final de fechamento da janela de visibilidade orbital.
    public DateTime EndTime { get; set; }

    // Por que: Largura de banda em Kbps suportada pelo link (ex: 1024.0 Kbps).
    public double BandwidthKbps { get; set; }

    // Por que: Latência fixa da velocidade da luz para a distância (ex: ~1.28s Terra-Lua).
    public double LatencySeconds { get; set; }

    // Por que: Status em tempo real do enlace de rádio (UP, DOWN, SCHEDULED).
    public string Status { get; set; } = string.Empty;
}
